packages/frontend/src/visualization/graphviz_svg.tsx

```diff
@@ -35,8 +36,18 @@ function GraphvizOutputSVG(props: {
     graph?: GraphvizJSON.Graph;
 }) {
     return (
-        <div class="graphviz">
+        <div class="graphviz" ref={visualizationRef}>
             <GraphSVG graph={props.graph && parseGraphvizJSON(props.graph)} />
         </div>
     );
 }
+
+// Create a ref for the visualization container
+let visualizationRef: HTMLDivElement | undefined;
+
+// export handling
+export const handleExportSVG = () => {
+    if (visualizationRef) {
+        exportVisualizationSVG(visualizationRef);
+    }
+};
```

Comment:
The use of a global variable for the ref and a global export callback should be a warning. In fact, this code will not work as expected if an analysis contains multiple visualizations (which is possible!).

The easiest way to fix this is to first move the export button from the top-level toolbar to the cell itself (perhaps in the `GraphSVG` component), then [forward the ref](https://docs.solidjs.com/concepts/refs#forwarding-refs) to the `svg` element up the hierarchy.